### ISSUE TITLE
class "FeatureCloud"  and class "TemplateAlignment" are void

### ISSUE THREAD

**Issue by xiqiaosang** [mattmongeon/pcl_auto_seg#1]
Hi, dear Matt Mongeon:
   I don't understand why you clear all the contents in class "FeatureCloud"  and class "TemplateAlignment". Are they useless in your application?

**Comment by mattmongeon** [comment]
Yes, the additional code was unnecessary for my application and only added
clutter, and I had a very targeted goal which meant my code could be very
targeted.

**Comment by xiqiaosang** [comment]
After checking your code for several times, I still don't understand this point:  Is the class "TemplateAlignment" from header file "pcl_auto_seg/template_alignment.h"? Or does it have any relationship with other files? If no, then the codes from line 194 to line 201 in file "seg_block.cpp" runs for nothing and can not give a correct result.  Similarly, I also don't understand how you create a cube in line 274 in file "seg_block.cpp" using the following code "    template_cloud1.makeCube(0.03);" since the class "FeatureCloud" in "pcl_auto_seg/feature_cloud.h" is also void.

BTW, running only "openni.launch" and your node "seg_block", I got the following error, "[pcl::SampleConsensusInitialAlignment::selectSamples] No valid sample found after 2025 iterations. Relaxing min_sample_distance_ to 0.025000".  This may help you understand my problem of using your codes.

I guess I must misunderstand something important of your codes, please give me a hint.
